client-core/src/transaction_builder/default_transaction_builder.rs

```diff
@@ -0,0 +1,441 @@
+use failure::ResultExt;
+use secstr::SecStr;
+
+use chain_core::init::coin::Coin;
+use chain_core::tx::data::attribute::TxAttributes;
+use chain_core::tx::data::input::TxoPointer;
+use chain_core::tx::data::output::TxOut;
+use chain_core::tx::data::Tx;
+use chain_core::tx::fee::{Fee, FeeAlgorithm};
+use chain_core::tx::witness::TxWitness;
+use chain_core::tx::TxAux;
+use client_common::{ErrorKind, Result};
+
+use crate::{TransactionBuilder, WalletClient};
+
+/// Default implementation of `TransactionBuilder`
```

Comment:
It'd be good to document the "default" behaviour briefly (the main idea), so that one doesn't need to dig it up from the code. I guess currently the behaviour is:
```
repeat until the fee is enough:
1. sort the UTXOs by amounts (`sort_by(|a, b| a.1.cmp(&b.1).reverse()` -- is it desc/asc? I can't tell without looking up sort_by+cmp refs)
2. take UTXOs until the target amount is reached and return a difference that's taken as the fee
``` ?